mainchain/blockchain/block_operations.go

```diff
@@ -140,17 +144,42 @@ func (bo *BlockOperations) CreateProposalBlock(
 // New calculated state root is validated against the root field in block.
 // Transactions, new state and receipts are saved to storage.
 func (bo *BlockOperations) CommitAndValidateBlockTxs(block *types.Block, lastCommit stypes.LastCommitInfo, byzVals []stypes.Evidence) ([]*types.Validator, common.Hash, error) {
+	var sumTxsByteLength int
 	vals, root, blockInfo, err := bo.commitBlock(block.Transactions(), block.Header(), lastCommit, byzVals)
 	if err != nil {
 		return nil, common.Hash{}, err
 	}
 
+	opStart := time.Now()
 	bo.saveBlockInfo(blockInfo, block)
 	bo.blockchain.DB().WriteHeadBlockHash(block.Hash())
 	bo.blockchain.DB().WriteTxLookupEntries(block)
 	bo.blockchain.DB().WriteAppHash(block.Height(), root)
 	bo.blockchain.InsertHeadBlock(block)
 
+	if metrics.Enabled {
```

Comment:
These kinds of metrics would be considered as heavy operations. Should we separate these to be independent from other legacy metrics (rpc, trie, tx_pool,...) @lewtran @thang14 ?